packages/validators/typescript/src/TypeScriptValidator.js

```diff
@@ -48,18 +51,25 @@ export default new Validator({
         ts.sys,
         baseDir,
       );
-
-      langServiceCache[configHash] = ts.createLanguageService(
-        new LanguageServiceHost(options.inputFS, ts, parsedCommandLine),
-        ts.createDocumentRegistry(),
+      const host = new LanguageServiceHost(
+        options.inputFS,
+        ts,
+        parsedCommandLine,
       );
+      langServiceCache[configHash] = {
+        host,
+        service: ts.createLanguageService(host, ts.createDocumentRegistry()),
+      };
     }
 
     if (!langServiceCache[configHash]) return;
 
-    const diagnostics = langServiceCache[configHash].getSemanticDiagnostics(
-      asset.filePath,
-    );
+    // Make sure that when the typescript language service asks us for this file, we let it know that there is a new version.
+    langServiceCache[configHash].host.invalidate(asset.filePath);
+
+    const diagnostics = langServiceCache[
```

Comment:
Prettier does some not so pretty things from time to time not sure if we can even do anything about it.